### PR TITLE
feat: migrateStrategy shouldn't reset activation time

### LINKED ISSUE #136: migrateStrategy shouldn't reset activation time
`migrateStrategy` shouldn't empty the replaced strategy params, it should do the same thing as revoke.

### PR DESCRIPTION
**Pull request by Macarse**
fixes #136

### CODE REVIEW

File: tests/functional/vault/test_strategies.py [hunk 1]
```diff
@@ -137,8 +137,9 @@ def test_migrateStrategy(gov, vault, strategy, rando, TestStrategy):
     new_strategy = gov.deploy(TestStrategy, vault)
     vault.migrateStrategy(strategy, new_strategy, {"from": gov})
 
-    # Can migrate back again (but it starts out fresh)
-    vault.migrateStrategy(new_strategy, strategy, {"from": gov})
+    # Can't migrate back again
```
Comment (Macarse) [inline tests/functional/vault/test_strategies.py:140]: since activation != 0, it can't be migrated back.

File: contracts/Vault.vy [hunk 2]
```diff
@@ -1117,6 +1117,13 @@ def updateStrategyPerformanceFee(
     log StrategyUpdatePerformanceFee(strategy, performanceFee)
 
 
+@internal
+def _revokeStrategy(strategy: address):
```
Comment (Macarse) [inline contracts/Vault.vy:1121]: I couldn't find a way to call an external method so I made the implementation `internal`

File: contracts/Vault.vy [hunk 3]
```diff
@@ -1140,7 +1147,7 @@ def migrateStrategy(oldVersion: address, newVersion: address):
     assert self.strategies[newVersion].activation == 0
 
     strategy: StrategyParams = self.strategies[oldVersion]
-    self.strategies[oldVersion] = empty(StrategyParams)
+    self._revokeStrategy(oldVersion)
     self.strategies[newVersion] = strategy
```
Comment (fubuloubu) [inline contracts/Vault.vy:1151]: This is probably not the best way to handle this, probably want to individually set the values, namely the parameters (don't copy the gains/losses) and total debt

Comment (Macarse) [reply contracts/Vault.vy:1151]: your point is that we would consider it a completely new strat?

Comment (fubuloubu) [reply contracts/Vault.vy:1151]: The past performance of the old strategy and the future performance of the new strategy should be treated separately

Comment (Macarse) [reply contracts/Vault.vy]: Did all the changes you requested but copied `totalDebt`.

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**